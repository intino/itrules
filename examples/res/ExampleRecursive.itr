def type(Module) AND attribute(modules)
	<module name="$name">
	    $modules...[$NL]
    </module>
end

def type(Module)
	<module name="$name"/>
end
