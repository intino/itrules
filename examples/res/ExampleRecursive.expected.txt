<module name="X">
    <module name="1">
        <module name="1.1"/>
        <module name="1.2">
            <module name="1.2.1"/>
            <module name="1.2.2"/>
            <module name="1.2.3"/>
        </module>
        <module name="1.3"/>
        <module name="1.4"/>
    </module>
    <module name="2">
        <module name="2.1"/>
        <module name="2.2"/>
        <module name="2.3"/>
        <module name="2.4"/>
        <module name="2.5"/>
        <module name="2.6"/>
    </module>
    <module name="3">
        <module name="3.1"/>
        <module name="3.2"/>
        <module name="3.3"/>
    </module>
    <module name="4">
        <module name="4.1"/>
        <module name="4.2"/>
    </module>
    <module name="5"/>
</module>